File: code-bugfixing/etl-processing/C#/VisualStudio/etlprocessing/CsvEtlCommand.cs
public class CsvEtlCommand
{
    private string sourceFile;
    private string targetFile;

    public CsvEtlCommand(string sourceFile, string targetFile)
    {
        this.sourceFile = sourceFile;
        this.targetFile = targetFile;
    }

    public void Execute()
    {
        using (var reader = new StreamReader(sourceFile))
        using (var writer = new StreamWriter(targetFile))
        {
            // Extract
            string line = reader.ReadLine();

            // Transform and Load
            while (line != null)
            {
                string[] columns = line.Split(',');
                var transformedLine = Transform(columns);

                writer.WriteLine(transformedLine);
                line = reader.ReadLine();
            }
        }
    }

    private string Transform(string[] columns)
    {
        // Add a simple transformation: reverse the order of the columns
        string transformed = "";
        for (int i = columns.Length - 1; i >= 0; i--)
        {
            transformed += columns[i];
            if (i > 0) transformed += ",";
        }
        return transformed;
    }
}

File: code-bugfixing/etl-processing/java/src/main/java/com/epam/poc/aicode/javabugfixing/etlprocessing/CsvEtlCommand.java
package com.epam.poc.aicode.javabugfixing.etlprocessing;

import java.io.*;

public class CsvEtlCommand {
    private String sourceFile;
    private String targetFile;

    public CsvEtlCommand(String sourceFile, String targetFile) {
        this.sourceFile = sourceFile;
        this.targetFile = targetFile;
    }

    public void execute() throws IOException {
        BufferedReader reader = null;
        BufferedWriter writer = null;

        try {
            reader = new BufferedReader(new FileReader(sourceFile));
            writer = new BufferedWriter(new FileWriter(targetFile));

            // Extract
            String line = reader.readLine();

            // Transform and Load
            while (line != null) {
                String[] columns = line.split(",");
                var transformedLine = transform(columns);

                writer.write(transformedLine + "\n");
                line = reader.readLine();
            }
        } finally {
            if (reader != null) reader.close();
            if (writer != null) writer.close();
        }
    }


    private String transform(String[] columns) {
        // Add a simple transformation: reverse the order of the columns
        String transformed = "";
        for (int i = columns.length - 1; i >= 0; i--) {
            transformed += columns[i];
            if (i > 0) transformed += ",";
        }
        return transformed;
    }
}
